controllers/restaurantController.js

```diff
@@ -0,0 +1,165 @@
+import { Restaurant } from "../models/Restaurant.js";
+import mongoose from "mongoose";
+
+const getRestaurants = async function (req, res) {
+  try {
+    const queryObj = { ...req.query };
+    let query = Restaurant.find(queryObj);
+    const restaurants = await query;
+
+    if (restaurants.length === 0) {
+      return res.status(404).json({
+        success: false,
+        error: "No restaurant found",
+      });
+    }
+
+    return res.status(200).json({
+      success: true,
+      data: restaurants,
+    });
+  } catch (err) {
+    return res.status(500).json({
+      success: false,
+      error: "Server Error",
+    });
+  }
+};
+
+const addRestaurant = async function (req, res) {
+  const restaurantToAdd = new Restaurant({
```

Comment:
please take a look in Agnieszka's PR, she has validation there. You can add validation here too